src/ssl.c

```diff
@@ -54957,15 +54956,45 @@ int wolfSSL_CONF_cmd(WOLFSSL_CONF_CTX* cctx, const char* cmd, const char* value)
     WOLFSSL_BIO *wolfSSL_BIO_new_connect(const char *str)
     {
         WOLFSSL_BIO *bio;
+        const char* port;
         WOLFSSL_ENTER("wolfSSL_BIO_new_connect");
         bio = wolfSSL_BIO_new(wolfSSL_BIO_s_socket());
         if (bio) {
-            bio->ip = str;
+            port = XSTRSTR(str, ":");
+
+            if (port != NULL)
+                bio->port = (word16)XATOI(port + 1);
+            else
+                port = str + XSTRLEN(str); /* point to null terminator */
+
+            bio->ip = (char*)XMALLOC(1 + port - str, bio->heap,
```

Comment:
Can you please rewrite as `(port - str) + 1` and add comment the `/* +1 is null termination */`

Reply:
Rewritten.